### DISCUSSION TITLE
not working example in Key Information Extraction (KIE) documentation

### DISCUSSION THREAD

**Discussion by goozzik** [PaddlePaddle/PaddleOCR#6914] [Q&A] [answered]
Hey.

I'm going through the steps described here https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.5/ppstructure/docs/kie_en.md and I'm stuck on `Download the pretrained model and predict the result:` part. It says that after running:

```
python3.7 tools/infer_kie.py -c configs/kie/kie_unet_sdmgr.yml -o Global.checkpoints=kie_vgg16/best_accuracy  Global.infer_img=../wildreceipt/1.txt
```

I should get some results in txt file:

> The prediction result is saved as ./output/sdmgr_kie/predicts_kie.txt, and the visualization results are saved in the folder/output/sdmgr_kie/kie_results/.

I'm executing the script and it finishes correctly, here is proof:

```
[2022/07/16 16:20:33] ppocr INFO: The Kie Image saved in ./output/sdmgr_kie/kie_results/1.png
[2022/07/16 16:20:37] ppocr INFO: The Kie Image saved in ./output/sdmgr_kie/kie_results/2.png
[2022/07/16 16:20:42] ppocr INFO: The Kie Image saved in ./output/sdmgr_kie/kie_results/3.png
[2022/07/16 16:20:42] ppocr INFO: success!
[2022/07/16 16:20:42] ppocr INFO: It took 15.519896030426025 s for predict 4 images.
[2022/07/16 16:20:42] ppocr INFO: The ips is 0.2577336853390118 images/s
➜  paddleOCR git:(release/2.5) ✗
```

but there is nothing in the text file:

```
➜  paddleOCR git:(release/2.5) ✗ cat ./output/sdmgr_kie/predicts_kie.txt
➜  paddleOCR git:(release/2.5) ✗
```

In `tools/infer_kie.py` it indeed opens `save_res_path`, which is our `./output/sdmgr_kie/predicts_kie.txt` in this case, but it never actually writes to it (you can see that there are no references to `fout`).

Can somebody advise? Am I doing something wrong? Or is this guide simply wrong?

Regards,
goozzik

**Comment by LDOUBLEV** [comment, marked as answer]
Please update the code to the latest, we just merged a PR recently that supports writing kie results to txt file. The PR: https://github.com/PaddlePaddle/PaddleOCR/pull/6824

**Reply by goozzik** [reply]
Thanks!
